src/GoTrueClient.ts

```diff
@@ -904,22 +904,25 @@ export default class GoTrueClient {
    * @param value time intervals in milliseconds.
    * @param session The current session.
    */
-  private _startAutoRefreshToken(value: number, session: Session) {
+  private _startAutoRefreshToken(value: number) {
     if (this.refreshTokenTimer) clearTimeout(this.refreshTokenTimer)
     if (value <= 0 || !this.autoRefreshToken) return
 
     this.refreshTokenTimer = setTimeout(async () => {
       this.networkRetries++
-      const { error } = await this._callRefreshToken(session.refresh_token)
-      if (!error) this.networkRetries = 0
-      if (
-        error instanceof AuthRetryableFetchError &&
-        this.networkRetries < NETWORK_FAILURE.MAX_RETRIES
-      )
-        this._startAutoRefreshToken(
-          NETWORK_FAILURE.RETRY_INTERVAL ** this.networkRetries * 100,
-          session
-        ) // exponential backoff
+      const {
+        data: { session },
+        error: sessionError,
+      } = await this.getSession()
+      if (!sessionError && session) {
+        const { error } = await this._callRefreshToken(session.refresh_token)
+        if (!error) this.networkRetries = 0
+        if (
+          error instanceof AuthRetryableFetchError &&
+          this.networkRetries < NETWORK_FAILURE.MAX_RETRIES
+        )
+          this._startAutoRefreshToken(NETWORK_FAILURE.RETRY_INTERVAL ** this.networkRetries * 100) // exponential backoff
```

Comment:
this seems error prone too - in the event of network failure, the retry interval could possibly fall outside of the reuse interval error

Reply:
Yeah, I agree with you that the user should be logged out here. We can always take another look at this if it becomes a bigger issue in the future.

Reply:
@kangmingtay, @alaister. A robust approach could be to listen for a network status change.
https://developer.mozilla.org/en-US/docs/Web/API/Navigator/onLine

This could affect users who are traveling and have their laptops offline for a day. Ideally, you'd want to stay logged in, it's an edge case, but a listener would capture these events. Then you can avoid long polling.